shared_model/validators/field_validator.hpp

```diff
@@ -99,6 +103,10 @@ namespace shared_model {
       void validateCounter(ReasonsGroupType &reason,
                            const interface::types::CounterType &counter) const;
 
+      void validateSignatures(ReasonsGroupType &reason,
```

Comment:
Why not to make `validateSignature` const and return `reason`?

Reply:
Then the whole API of validator should be changed . Here we modify existing reason by including new errors if any, so passing by reference makes sense